plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+
+            if 'servers' in want and want['servers'] is not None:
+                for server in want['servers']:
+                    server.pop('key_id')
+                    server.pop('minpoll')
+                    server.pop('maxpoll')
+            if 'ntp_keys' in want and want['ntp_keys'] is not None:
+                for ntp_key in want['ntp_keys']:
+                    ntp_key.pop('encrypted')
+                    ntp_key.pop('key_type')
+                    ntp_key.pop('key_value')
+
+        elif state == 'merged':
+            if 'servers' in want and want['servers'] is not None:
+                for server in want['servers']:
+                    if not server['key_id']:
+                        server.pop('key_id')
+                    if not server['minpoll']:
+                        server.pop('minpoll')
+                    if not server['maxpoll']:
+                        server.pop('maxpoll')
+
+    def get_create_requests(self, configs, have):
+
+        requests = []
+
+        enable_auth_config = configs.get('enable_ntp_auth', None)
+        if enable_auth_config is not None:
+            enable_auth_request = self.get_create_enable_ntp_auth_requests(enable_auth_config, have)
+            if enable_auth_request:
+                requests.extend(enable_auth_request)
+
+        src_intf_config = configs.get('source_interfaces', None)
+        if src_intf_config:
+            src_intf_request = self.get_create_source_interface_requests(src_intf_config, have)
+            if src_intf_request:
+                requests.extend(src_intf_request)
+
+        keys_config = configs.get('ntp_keys', None)
+        if keys_config:
+            keys_request = self.get_create_keys_requests(keys_config, have)
+            if keys_request:
+                requests.extend(keys_request)
+
+        servers_config = configs.get('servers', None)
+        if servers_config:
+            servers_request = self.get_create_servers_requests(servers_config, have)
+            if servers_request:
+                requests.extend(servers_request)
+
+        trusted_key_config = configs.get('trusted_keys', None)
+        if trusted_key_config:
+            trusted_key_request = self.get_create_trusted_key_requests(trusted_key_config, have)
+            if trusted_key_request:
+                requests.extend(trusted_key_request)
+
+        vrf_config = configs.get('vrf', None)
+        if vrf_config:
+            vrf_request = self.get_create_vrf_requests(vrf_config, have)
+            if vrf_request:
+                requests.extend(vrf_request)
+
+        return requests
+
+    def preprocess_delete_commands(self, commands, have):
+
+        new_commands = {}
+
+        enable_auth_command = commands.get('enable_ntp_auth', None)
+        enable_auth_have = have.get('enable_ntp_auth', None)
+        if enable_auth_command is not None and enable_auth_have is not None and \
+           (enable_auth_command or enable_auth_have):
```

Comment:
The deletion should only occur if the two values are the same and is a no-op unless the "and" of these values is True.

Because of this, the line above should be:

```
(enable_auth_command and enable_auth_have)
```

Reply:
This is for regression test.

Reply:
The regression should pass using the more efficient "and" criterion. (This avoids unnecessary command accesses to the switch.)

Reply:
Removed preprocess_delete_commands function.

Reply:
Verified.